### PR TITLE
transport: enable keep-alives and limit number of sockets

### PR DESCRIPTION
**Pull request by mattrobenolt**

### CODE REVIEW

**Comment by mattrobenolt** [comment]
I need to confirm yet this actually does what I think it does.

File: lib/transports.js [hunk 1]
```diff
@@ -6,12 +6,16 @@ var util = require('util');
 function Transport() {}
 util.inherits(Transport, events.EventEmitter);
 
+var agentOptions = { keepalive: true, maxSockets: 100 };
```
Comment (LewisJEllis) [inline lib/transports.js]: 100 seems high to me? Also, should we provide a `maxFreeSockets` option? (https://nodejs.org/api/http.html#http_new_agent_options)

File: lib/transports.js [hunk 2]
```diff
@@ -6,12 +6,16 @@ var util = require('util');
...
+
 var http = require('http');
+var httpAgent = new http.Agent(agentOptions);
 
 function HTTPTransport(options) {
   this.defaultPort = 80;
   this.transport = http;
   this.options = options || {};
+  this.agent = httpAgent;
```
Comment (LewisJEllis) [inline lib/transports.js]: We could theoretically enable users to change agent options via something like:
```javascript
    Raven.config('___DSN___', {
      transport: new raven.transports.HTTPSTransport({agentOpts: { ... }})
    });
```
but I'm not sure whether we need to or should; are there cases where users might want to do this (or we might want them to)? Guessing you have more background on our intentions behind transports being pluggable/user-configurable.

**Comment by benvinegar** [comment]
*ping*

File: lib/transports.js (hunk 1 above)
Comment (kamilogorek) [reply lib/transports.js]: Number of `maxFreeSockets` won't go over `maxSockets` number and it defaults to `256` so we should be good here.

File: lib/transports.js (hunk 2 above)
Comment (kamilogorek) [reply lib/transports.js]: Agree on enabling user to change those settings.

**Comment by kamilogorek** [comment]
Ping @mattrobenolt

File: lib/transports.js (hunk 2 above)
Comment (kamilogorek) [reply lib/transports.js]: + adding a comment in the docs about it

**Comment by mattrobenolt** [comment]
> I need to confirm yet this actually does what I think it does.

It does. This would be nice to have.

**Comment by kamilogorek** [comment]
@benvinegar @mattrobenolt updated. First two commits come from https://github.com/getsentry/raven-node/pull/369 as it's related, and it should be merged first.

**Comment by kamilogorek** [comment]
Ping

**Comment by kamilogorek** [comment]
I just realised that this code has been partially broken due to incorrect socket name being constructed. Fixed and updated test.

This should also fix https://github.com/getsentry/raven-node/issues/214 enough for it to be closed.

**Pull request merged**